### DISCUSSION TITLE
RunTrigger event safeguard

### DISCUSSION THREAD

**Discussion by JeremyVyska** [microsoft/alguidelines#201] [BC Best Practices]
There are a small variety of events in the system that involve an optional "RunTigger" pass.

Doing a check on this in your template is as important (IMHO) as both the isTemporary safeguard and the IsHandled pattern.

(This discussion is mostly for me to remember that we need this and to write it this fall, but anyone is welcome to beat me to it.)

**Comment by SorenKlemmensen** [comment]
Good point in regard to isTemporary. It is one of my code review guidelines to make sure this is considered, and I cannot think of any time that it should not be done. As for the Handled Pattern I am not sure yet. I would agree it should be considered however I could see things like permissions or license rights that you would not want people to write their own code and bypass yours. Anyway, this is great stuff to think about.
